pelicun/model/loss_model.py

```diff
@@ -1210,72 +1241,101 @@ def _aggregate_sample(self, sample: pd.DataFrame, columns: list) -> pd.DataFrame
+            columns=replacement_consequence_RV_reg.RV.keys(),
             dtype=bool,
         )
 
-        # If there is no `replacement` in the sample, simply return.
-        if 'replacement' not in sample.columns.get_level_values('loss'):
-            return sample, exceedance_bool_df
+        # Sample replacement consequences from the registry
+        replacement_consequence_RV_reg.generate_sample(len(sample), 'MonteCarlo')
+
+        sample_dvs = replacement_consequence_RV_reg.RV.keys()
+        for sample_dv in sample_dvs:
+            sub_sample = sample.loc[:, sample_dv]
+            if 'replacement' in sub_sample.columns.get_level_values('loss'):
+                # If `replacement` already exists as a consequence,
+                # determine the realizations where it is non-zero.
+                no_replacement_mask = (
+                    ~(sub_sample['replacement'] > 0.00).any(axis=1).values
+                )
+                no_replacement_columns = (
+                    sub_sample.columns.get_level_values('loss') != 'replacement'
+                )
+            else:
+                # Otherwise there is no row where we already have replacement
+                no_replacement_mask = np.full(len(sub_sample), True)
+                no_replacement_columns = np.full(len(sub_sample.columns), True)
+            # Get the sum to compare with the thresholds
+            consequence_sum_given_no_replacement = sub_sample.iloc[
+                no_replacement_mask, no_replacement_columns
+            ].sum(axis=1)
+            if not consequence_sum_given_no_replacement.empty:
+                consequence_values = replacement_consequence_RV_reg.RV[
+                    sample_dv
+                ].sample
+                exceedance_mask = (
+                    consequence_sum_given_no_replacement
+                    > consequence_values[no_replacement_mask]
+                    * replacement_ratios[sample_dv]
+                )
+                # Fill the remaining rows with False
+                if len(exceedance_mask) < len(sub_sample):
+                    exceedance_mask = exceedance_mask.reindex(
+                        sub_sample.index, fill_value=False
+                    )
+            else:
+                exceedance_mask = pd.Series(
+                    np.full(len(sub_sample), False), index=sub_sample.index
+                )
 
-        replacement_realizations = {}
+            # Monitor triggering of replacement
+            exceedance_bool_df[sample_dv] = exceedance_mask
+
+        # Assign replacement consequences where the threshold has been
+        # exceeded.
+        exceedance_realizations = exceedance_bool_df.any(axis=1)
+        # Assign replacement consequences: needs to include all DVs
+        for other_dv in replacement_consequence_RV_reg.RV.keys():
+            col = (
+                other_dv,
+                'replacement',
+                'threshold_exceedance',
```

Comment:
I like that we can put the "triggering event: threshold_exceedance" here and it is one more reason to keep both the dmg and loss levels.

Reply:
I agree it is better to keep that level.